pkg/errors/factory.go

```diff
@@ -47,16 +47,11 @@ func GetError(err error) error {
 		return nil
 	}
 
+	var errAPIStatus api_errors.APIStatus
 	switch {
 	case isEmptyConfigError(err):
 		return newNoKubeConfig(err.Error())
-	case isNoRouteToHostError(err):
-		return newNoRouteToHost(err.Error())
-	default:
-		var errAPIStatus api_errors.APIStatus
-		if !errors.As(err, &errAPIStatus) {
-			return err
-		}
+	case errors.As(err, &errAPIStatus):
```

Comment:
I'd suggest to add 2 new functions `isStatusError()` and then `newStatusError()` to wrap the handling of that case into it. I think it'd make the cases more self-explanatory.